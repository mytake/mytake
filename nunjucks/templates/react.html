{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
