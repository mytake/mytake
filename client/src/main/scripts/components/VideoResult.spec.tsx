import * as React from "react";
import * as renderer from "react-test-renderer";
import VideoResult from "./VideoResult";
import { kennedyNixon } from "../utils/testUtils";

jest.mock("./VideoResultTurnList", () => ({
  default: "VideoResultTurnList"
}));

test("VideoResultPreview containing", () => {
  const tree = renderer
    .create(
      <VideoResult
        multiHighlight={{
          cut: [14, 239],
          highlights: [[18, 28]]
        }}
        onPlayClick={jest.fn()}
        sortBy="Containing"
        turn={0}
        turnContent="Good evening. The television and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction. The Republican candidate, Vice President Richard M. Nixon, and the Democratic candidate, Senator John F. Kennedy. According to rules set by the candidates themselves, each man shall make an opening statement of approximately eight minutes' duration and a closing statement of approximately three minutes' duration. In between the candidates will answer, or comment upon answers to questions put by a panel of correspondents. In this, the first discussion in a series of four - joint appearances, the subject matter has been agreed, will be restricted to internal or domestic American matters. And now for the first opening statement by Senator John F. Kennedy."
        videoFact={kennedyNixon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("VideoResultPreview before and after", () => {
  const tree = renderer
    .create(
      <VideoResult
        multiHighlight={{
          cut: [0, 276],
          highlights: [[18, 28]]
        }}
        onPlayClick={jest.fn()}
        sortBy="BeforeAndAfter"
        turn={0}
        turnContent="Good evening. The television and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction. The Republican candidate, Vice President Richard M. Nixon, and the Democratic candidate, Senator John F. Kennedy. According to rules set by the candidates themselves, each man shall make an opening statement of approximately eight minutes' duration and a closing statement of approximately three minutes' duration. In between the candidates will answer, or comment upon answers to questions put by a panel of correspondents. In this, the first discussion in a series of four - joint appearances, the subject matter has been agreed, will be restricted to internal or domestic American matters. And now for the first opening statement by Senator John F. Kennedy."
        videoFact={kennedyNixon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
