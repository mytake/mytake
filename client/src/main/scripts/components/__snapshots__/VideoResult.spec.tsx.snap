// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoResultPreview before and after 1`] = `
<div
  className="turn"
>
  <div
    className="turn__info"
  >
    <h3
      className="turn__speaker"
    >
       Smith
    </h3>
    <p
      className="turn__time"
    >
      0:1 - 0:17
    </p>
    <div
      className="turn__actions"
    >
      <button
        className="turn__button turn__button--play"
        onClick={[Function]}
      >
        Play
      </button>
      <button
        className="turn__button turn__button--open"
        onClick={[Function]}
      >
        Open
      </button>
    </div>
  </div>
  <p
    className="turn__results"
  >
    Good evening. The 
    <strong>
      television
    </strong>
     and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction.
  </p>
</div>
`;

exports[`VideoResultPreview containing 1`] = `
<div
  className="turn"
>
  <div
    className="turn__info"
  >
    <h3
      className="turn__speaker"
    >
       Smith
    </h3>
    <p
      className="turn__time"
    >
      0:2 - 0:15
    </p>
    <div
      className="turn__actions"
    >
      <button
        className="turn__button turn__button--play"
        onClick={[Function]}
      >
        Play
      </button>
      <button
        className="turn__button turn__button--open"
        onClick={[Function]}
      >
        Open
      </button>
    </div>
  </div>
  <p
    className="turn__results"
  >
    The 
    <strong>
      television
    </strong>
     and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency.
  </p>
</div>
`;
